models/issues/comment.go

```diff
@@ -181,40 +182,40 @@ func (t CommentType) HasAttachmentSupport() bool {
 	return false
 }
 
+type Role string
+
 // RoleDescriptor defines comment tag type
-type RoleDescriptor int
+type RoleDescriptor struct {
+	IsPoster util.OptionalBool
+	Role     Role
+}
 
 // Enumerate all the role tags.
 const (
-	RoleDescriptorNone RoleDescriptor = iota
-	RoleDescriptorPoster
-	RoleDescriptorWriter
-	RoleDescriptorOwner
+	RoleDescriptorOwner                Role = "owner"
+	RoleDescriptorMember               Role = "member"
+	RoleDescriptorCollaborator         Role = "collaborator"
+	RoleDescriptorFirstTimeContributor Role = "first_time_contributor"
+	RoleDescriptorContributor          Role = "contributor"
 )
 
-// WithRole enable a specific tag on the RoleDescriptor.
-func (rd RoleDescriptor) WithRole(role RoleDescriptor) RoleDescriptor {
-	return rd | (1 << role)
+// HasRole returns if a role is not none
+func (r Role) HasRole() bool {
+	return r.String() != ""
 }
 
-func stringToRoleDescriptor(role string) RoleDescriptor {
-	switch role {
-	case "Poster":
-		return RoleDescriptorPoster
-	case "Writer":
-		return RoleDescriptorWriter
-	case "Owner":
-		return RoleDescriptorOwner
-	default:
-		return RoleDescriptorNone
-	}
+func (r Role) String() string {
+	return string(r)
+}
+
+// LocaleString returns the locale string name of the Status
+func (r Role) LocaleString(lang translation.Locale) string {
+	return lang.Tr("repo.issues." + r.String())
```

Comment:
It needs some prefix like `repo.issues.role.` ...

Otherwise it's strange to mix with other strings like `repo.issues.ref_from`

Reply:
Done.